### PR TITLE
DEPR: remove deprecated verbose, private_key kwargs for gbq

### PR DESCRIPTION
**Pull request by jbrockmendel**
The deprecation didn't have a whatsnew note or actually use deprecate_kwarg, so im not sure how formal the removal should be.

xref #20564.

### CODE REVIEW

**Comment by jreback** [comment]
need to remove these tests too: > test_read_gbq_with_deprecated_kwargs

**Comment by jbrockmendel** [comment]
removed test, added a whatsnew note that didnt make it into #30199

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -534,6 +534,7 @@ or ``matplotlib.Axes.plot``. See :ref:`plotting.formatters` for more.
 - :meth:`Series.plot` no longer accepts positional arguments, pass keyword arguments instead (:issue:`30003`)
 - :meth:`DataFrame.hist` and :meth:`Series.hist` no longer allows ``figsize="default"``, specify figure size by passinig a tuple instead (:issue:`30003`)
 - Floordiv of integer-dtyped array by :class:`Timedelta` now raises ``TypeError`` (:issue:`21036`)
+- :class:`TimedeltaIndex` and :class:`DatetimeIndex` no longer accept non-nanosecond dtype strings like "timedelta64" or "datetime64", use "timedelta64[ns]" and "datetime64[ns]" instead (:issue:`24806`)
```
Comment (tswast) [inline doc/source/whatsnew/v1.0.0.rst]: Doesn't seem related to this change.

**Review by tswast — APPROVED** [review]

File: doc/source/whatsnew/v1.0.0.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.0.0.rst]: correct, see comment https://github.com/pandas-dev/pandas/pull/30200#issuecomment-564635679

**Comment by jreback** [comment]
IIRC I remember seeing something about ``private_key`` still in pandas-gbq? @tswast

**Comment by tswast** [comment]
That's right. I filed https://github.com/pydata/pandas-gbq/issues/301 to actually remove it. I'll do that and cut a pandas-gbq release today.

**Comment by jbrockmendel** [comment]
so should i revert the private_key change here?  warn?

**Comment by tswast** [comment]
It's okay to remove. We've been showing a warning since we deprecated it in pandas 0.24.0. Soon it'll hard fail if it's ever supplied.

**Review by WillAyd — COMMENTED** [review]
Looks good. Can you add a whatsnew for this change?

**Comment by WillAyd** [comment]
Thanks @jbrockmendel
